### ISSUE TITLE
Incorrect "Unused LANGUAGE pragma" lint when using `TypeApplications` at the type level

### ISSUE THREAD

**Issue by TotallyNotChase** [ndmitchell/hlint#1343]
This file-
```hs
{-# LANGUAGE RankNTypes #-}
{-# LANGUAGE GADTs #-}
{-# LANGUAGE PolyKinds #-}
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE DataKinds #-}

module Main (main) where
import Data.Kind (Type)

data Foo :: forall (r :: Type). Type where
    MkFoo :: Integer -> a -> Foo @r
```
produces-
```
Unused LANGUAGE pragma
Found:
  {-# LANGUAGE TypeApplications #-}
Why not:
  
hlint(refact:Unused LANGUAGE pragma)
```
However, removing `TypeApplications` fails compilation, as the type application has indeed been used in the GADT declaration.

This also applies to function type signatures-
```hs
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE DataKinds #-}

x :: Foo @Int
```
also complains about the `TypeApplications` pragma being unused.

**Comment by shayne-fletcher** [comment]
i believe this is fixed by https://github.com/ndmitchell/hlint/pull/1335

**Comment by TotallyNotChase** [comment]
Ah, I see - it does seem like a duplicate of #1334. Very nice, will check it out on next release!
